src/admin/index.js

```diff
@@ -620,6 +619,73 @@ module.exports = ({
     return { groups }
   }
 
+  /**
+   * List groups in a broker
+   *
+   * @param {string[]} [groupIds]
+   * @return {Promise<DeleteGroups>}
+   *
+   * @typedef {Object} DeleteGroups
+   * @property {Array<DeleteGroup>} results
+   *
+   * @typedef {Object} DeleteGroup
+   * @property {string} groupId
+   * @property {number} errorCode
+   */
+  const deleteGroups = async groupIds => {
+    if (!groupIds || !Array.isArray(groupIds)) {
+      throw new KafkaJSNonRetriableError(`Invalid groupIds array ${groupIds}`)
+    }
+
+    const invalidGroupId = groupIds.some(g => typeof g !== 'string')
+
+    if (invalidGroupId) {
+      throw new KafkaJSNonRetriableError(`Invalid groupId name: ${JSON.stringify(invalidGroupId)}`)
+    }
+
+    const retrier = createRetry(retry)
+
+    return retrier(async (bail, retryCount, retryTime) => {
+      try {
+        await cluster.refreshMetadata()
+
+        const brokersPerGroups = {}
+        const brokers = {}
+        for (const groupId of groupIds) {
+          const broker = await cluster.findGroupCoordinator({ groupId })
+          if (brokersPerGroups[broker.nodeId] === undefined) brokersPerGroups[broker.nodeId] = []
+          brokersPerGroups[broker.nodeId].push(groupId)
+          brokers[broker.nodeId] = broker
+        }
+
+        const promises = []
+        for (const nodeId in brokersPerGroups) {
+          promises.push(brokers[nodeId].deleteGroups(brokersPerGroups[nodeId]))
+        }
+
+        if (promises.length === 0) return { results: [] }
+
+        const response = await Promise.all(promises)
+
+        const results = []
```

Comment:
This could be simplified a bit to just `flatten(responses.map(({ results }) => results))` (`flatten` is in src/utils/flatten.js)

Reply:
Not required any more as code changed to be able to handle retries